src/conspiracies/prompt_relation_extraction/template_class.py

```diff
@@ -0,0 +1,303 @@
+from abc import abstractmethod
+from typing import List, Tuple, Optional, Dict
+from spacy.tokens import Doc
+from conspiracies.prompt_relation_extraction.data_classes import SpanTriplet
+
+
+class Template:
+    """Abstract class for template classes. The class is used to create a
+    template for the triplet extraction task. The template is used to create a
+    prompt for the GPT-3 model.
+
+    Methods:
+        generate_prompt(target_tweet, generate_string):
+            creates a prompt for the GPT-3 model with the given target tweet. boolean option to generate the string over
+        set_examples(examples):
+            sets the examples to be used in the prompt
+        set_task_description(task_description):
+            sets the task description to be used in the prompt
+    """
+
+    def __init__(
+        self,
+        examples: List[Dict[str, str]],
+        task_description: str,
+    ):
+        self.examples = examples
+        self.task_description = task_description
+        self.spacy_examples_to_list()
+        self.generate_prompt(generate_string=True)
+
+    def set_examples(self, examples: List[Tuple[Doc, List[SpanTriplet]]]):
+        self.examples = examples
+        self.spacy_examples_to_list()
+        self.generate_prompt(generate_string=True)
+
+    def set_task_description(self, task_description: str):
+        self.task_description = task_description
+        self.spacy_examples_to_list()
+        self.generate_prompt(generate_string=True)
+
+    @staticmethod
+    def get_list_triplets(triplets: List[SpanTriplet]):
+        return [
+            [triplet.subject.text, triplet.predicate.text, triplet.object.text]
+            for triplet in triplets
+        ]
+
+    def spacy_examples_to_list(self):
+        self.non_spacy = [
+            (example["doc"].text, self.get_list_triplets(example["triplets"]))
+            for example in self.examples
+        ]
+
+    @abstractmethod
+    def generate_prompt(self):
+        pass
```

Comment:
We should also add something like a `parse_prompt_output` function here as well.